bitcoin-fundamentals-review.asciidoc

```diff
@@ -0,0 +1,223 @@
+
+Atomicity:: You can make a secret pre-image a condition of spending funds in several linked transactions. If any one of the parties reveals the pre-image in order to spend one of the transactions, all the other parties can now spend their transactions too. All or none become spendable, achieving atomicity across several transactions.
+e to alter the message and still have the same hash.
+
+==== Digital signatures
+
+The private key is used to create signatures that are required to spend bitcoin by proving ownership of funds used in a transaction.
+
+A digital signature is a number that is calculated from the application of the private key to a specific message.
+
+Given a message m and a private key k, a signature function F_sig_ can produce a signature S:
+
+latexmath:[ S = F{sign}(m, k) ]
+
+This signature S can be independently verified by anyone who has the public key K (corresponding to private key k), and the message:
+
+latexmath:[ S' = F{verify}(m, K, S) ]
+
+If S' matches S, then the verifier can confirm that the message m was signed by someone who had access to the private key k. Importantly, the digital signature proves the possession of the private key k at the time of signing, without revealing k.
+
+Digital signatures use a cryptographic hash algorithm. The signature is applied to a hash of the message, so that the message m is "summarized" to a fixed-length hash H(m) that serves as a fingerprint.
+
+=== Bitcoin transactions
+
+Transactions are data structures that encode the transfer of value between participants in the bitcoin system.
+
+==== Inputs and outputs
+
+The fundamental building block of a bitcoin transaction is a transaction output. Transaction outputs are indivisible chunks of bitcoin currency, recorded on the blockchain, and recognized as valid by the entire network. A transaction spends "inputs" and creates "outputs". Transaction inputs are simply references to outputs of previously recorded transactions. This way, each transaction spends the outputs of a previous transactions and creates new outputs.
+
+[[transaction_structure]]
+.A transaction transfers value from inputs to outputs
+image::images/tx1.png["transaction inputs and outputs"]
+
+Bitcoin full nodes track all available and spendable outputs, known as _unspent transaction outputs_, or UTXO. The collection of all UTXO is known as the UTXO set and currently numbers in the millions of UTXO. The UTXO set grows as new UTXO is created and shrinks when UTXO is consumed. Every transaction represents a change (state transition) in the UTXO set, by consuming one or more UTXO as _transaction inputs_ and creating one or more UTXO as its _transaction outputs_.
```

Comment:
I was confused that "UTXO" is used as the plural for UTXO here. My understanding is that the plural of initialisms in English is always formed by appending a lowercase "s". When the initialism ends in a capital "S" it may be offset by an apostrophe. For once, the Brits and Americans seem to agree on something, the [Chicago Manual of Style](https://www.chicagomanualofstyle.org/qanda/data/faq/topics/Plurals/faq0013.html), [Lexico.com](https://www.lexico.com/grammar/initialisms), [onlinewritingtraining.com.au](https://onlinewritingtraining.com.au/acronyms-initialisms/), [everywordcounts.co.uk](https://www.everywordcounts.co.uk/plural-acronyms-apostrophe-no-apostrophe/) as well as a number of other sources I've looked at are in agreement here.

I would also argue that it makes it much easier to read a passage that mixes singular UTXO and plural UTXOs mentions.

Reply:
I agree that UTXO's is easier to read

Reply:
Fixed